test/storefronts/sync-all.test.ts

```diff
@@ -0,0 +1,90 @@
+import * as dotenv from 'dotenv'
+import axios from 'axios'
+import MockAdapter from 'axios-mock-adapter'
+import { v0 } from '../../src/tillhub-js'
+import { initThInstance } from '../util'
+dotenv.config()
+
+const legacyId = '4564'
+
+const mock = new MockAdapter(axios)
+afterEach(() => {
+  mock.reset()
+})
+
+const storeId = '1234'
+const storefrontsSyncResponse = {
+  msg: 'nice!'
+}
+
+describe('v0: Storefronts: can start the sync all products process', () => {
+  it("Tillhub's products are instantiable", async () => {
+    if (process.env.SYSTEM_TEST !== 'true') {
+      mock.onPost('https://api.tillhub.com/api/v0/users/login').reply(() => {
```

Comment:
maybe this mock can be moved up to afterEach, turning afterEach to beforeEach. As is used in the all cases?

Reply:
It's not really the case, because both the login and the tested call on the test are wrapped with a condition, the tested calls on each test are different. So unfortunately it's not possible.

Also, in general, the whole SDK codebase is tons of copy-pastes, so changing something (which will be really great if that get reduce boilerplate and increase readability) will require more work to make the changes across all similar instances in the codebase. Just something to bear in mind.